src/ecs/system/local/user_gateway.rs

```diff
@@ -205,6 +237,50 @@ fn handle_user_despawn(
     Ok(())
 }
 
+fn handle_return_to_lobby(
+    connection_global_world_id: EntityId,
+    connection_local_world_id: EntityId,
+    connections: &mut ViewMut<LocalConnection>,
+) -> Result<()> {
+    debug!("Message::RequestReturnToLobby incoming");
+
+    let connection = connections
+        .try_get(connection_local_world_id)
+        .context(format!(
+            "Can't find connection {:?}",
+            connection_local_world_id
+        ))?;
+    Ok(task::block_on(async {
+        send_message(
```

Comment:
You don't need to spawn a task here with ```block_on()```. I think you copied the code from ```handle_topo_fin()```, which was a copy/paste job when I moved the code that included database code from the global ECS to the local ECS.

```rust
    send_message(
        assemble_return_to_lobby(connection_global_world_id, connection_local_world_id),
        &connection.channel,
    );
    Ok(())
```